src/qibo/tests/test_core_circuit_noise.py

```diff
@@ -160,3 +160,31 @@ def test_density_matrix_circuit_errors():
     c = Circuit(5)
     with pytest.raises(ValueError):
         c.add(gates.KrausChannel([((0,), np.eye(2))]))
+
+
+def test_circuit_add_sampling():
```

Comment:
I would add the backend fixture here
```suggestion
def test_circuit_add_sampling(backend):
```
so that this is tested on all available backends. Right now it is only tested on the default backend and will have different behavior depending on your environment. In fact it fails for some backends (see comment below).

Reply:
Yep, you are right. It is better to test it for every backend.